generator/views.py

```diff
@@ -91,11 +119,289 @@ def from_db_views(
+                    ],
+                }
+            ]
+            view_defn["sql_table_name"] = "`{% parameter channel %}`"
+        else:
+            view_defn["sql_table_name"] = f"`{table}`"
+
+        return [view_defn]
+
+    def get_measures(self, dimensions: List[dict], table: str) -> List[Dict[str, str]]:
+        """Generate measures from a list of dimensions.
+
+        When no dimension-specific measures are found, return a single "count" measure.
+
+        Raise ClickException if dimensions result in duplicate measures.
+        """
+        measures = {}
+
+        for dimension in dimensions:
+            dimension_name = dimension["name"]
+            if dimension_name in {"client_id", "client_info__client_id"}:
+                measure = {
+                    "name": "clients",
+                    "type": "count_distinct",
+                    "sql": f"${{{dimension_name}}}",
+                }
+            elif dimension_name == "document_id":
+                measure = {"name": "ping_count", "type": "count"}
+            else:
+                continue
+            name = measure["name"]
+            if name in measures:
+                raise click.ClickException(
+                    f"duplicate measure {name!r} for table {table!r}"
+                )
+            measures[name] = measure
+
+        if len(measures) == 0:
+            raise click.ClickException(
+                f"Missing client_id and doc_id dimensions in {table!r}"
+            )
+
+        return list(measures.values())
+
 
 class GrowthAccountingView(View):
     """A view for growth accounting measures."""
 
     type: str = "growth_accounting_view"
+    other_dimensions: List[Dict[str, str]] = [
+        {
+            "name": "first",
+            "sql": "{TABLE}.first",
+            "type": "yesno",
+            "hidden": "yes",
+        }
+    ]
+
+    default_dimensions: List[Dict[str, str]] = [
+        {
+            "name": "active_this_week",
+            "sql": "mozfun.bits28.active_in_range(days_seen_bits, -6, 7)",
+            "type": "yesno",
+            "hidden": "yes",
+        },
+        {
+            "name": "active_last_week",
+            "sql": "mozfun.bits28.active_in_range(days_seen_bits, -13, 7)",
+            "type": "yesno",
+            "hidden": "yes",
+        },
+        {
+            "name": "new_this_week",
+            "sql": (
+                "DATE_DIFF(${submission_date}," "first_run_date, DAY) BETWEEN 0 AND 6"
+            ),
+            "type": "yesno",
+            "hidden": "yes",
+        },
+        {
+            "name": "new_last_week",
+            "sql": (
+                "DATE_DIFF(${submission_date}," "first_run_date, DAY) BETWEEN 7 AND 13"
+            ),
+            "type": "yesno",
+            "hidden": "yes",
+        },
+    ]
+
+    default_measures: List[Dict[str, Union[str, Dict[str, str]]]] = [
+        {
+            "name": "overall_active_previous",
+            "type": "count",
```

Comment:
either this view needs to define `client_id` as a primary key, or all these count measures need to use `count_distinct`, e.g.:
```suggestion
            "type": "count_distinct",
            "sql": "client_id",
```

Reply:
We definitely want to avoid `count_distinct` for these measures, where possible. I'm fine with adding `client_id` - `submission_date` as primary key.